LeetCodeSolutions/Dota2Senate/Dota2Senate.cs

```diff
@@ -0,0 +1,140 @@
+namespace LeetCodeSolutions;
+public class Dota2SenateSolution {
+    private enum Party{
+        Dire,
+        Radiant
+    }
+
+    private enum SenatorAction{
+        AnnounceVictory,
+        BanOpponent
+    }
+
+
+    private class Senator{
+        private readonly Party party;
+        public Senator(char initial){
+            switch(initial){
+                case 'D':
+                    party = Party.Dire;
+                    break;
+                case 'R':
+                    party = Party.Radiant;
+                    break;
+            }
+        }
+
+        
+        public SenatorAction ExerciseRight(Dictionary<Party, int> activeSenateCount){
+            if(activeSenateCount[party] > 0 && activeSenateCount[OpposingParty()] == 0){
+                return SenatorAction.AnnounceVictory;
+            }else{
```

Comment:
in retrospect, break early return reads just as well as an else statement. Don't know if it's worth changing